File: dataset/human/java/file_9353.java
import java.io.*;
import java.util.*;

import static java.lang.Integer.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader in = new BufferedReader(new InputStreamReader(System.in));
        PrintWriter pw = new PrintWriter(System.out);
        StringTokenizer st;
        int t = parseInt(in.readLine());
        int cnt = 0;
        for (int i = 0; i < t; i++) {
            st = new StringTokenizer(in.readLine());
            int D1 = parseInt(st.nextToken()), D2 = parseInt(st.nextToken());
            if (D1 == D2) cnt++;
            else if (cnt < 3) cnt = 0;

        }
        if (cnt >= 3) pw.println("Yes");
        else pw.println("No");
        pw.close();
    }

    static void solve() {

    }
}

File: dataset/human/cpp/file_9353.cpp
#include<cstdio>
using namespace std;

int main()
{
    int n,m;
    scanf("%d%d",&n,&m);
    char a[51][51],b[51][51];
    for (int i=1;i<=n;i++)
    {
        scanf("\n");
        for (int j=1;j<=n;j++)
        {
            scanf("%c",&a[i][j]);
        }
    }
    for (int i=1;i<=m;i++)
    {
        scanf("\n");
        for (int j=1;j<=m;j++)
        {
            scanf("%c",&b[i][j]);
        }
    }
    for (int x=1;x<=n-m+1;x++)
    {
        for (int y=1;y<=n-m+1;y++)
        {
            bool ok=true;
            for (int i=1;i<=m;i++)
            {
                for (int j=1;j<=m;j++)
                {
                    if (b[i][j]!=a[x+i-1][y+j-1])
                    {
                        ok=false;
                    }
                }
            }
            if (ok)
            {
                printf("Yes\n");
                return 0;
            }
        }
    }
    printf("No\n");
    return 0;
}
